source/computational_domain.cc

```diff
@@ -589,9 +599,8 @@ void ComputationalDomain<dim>::refine_and_resize(const unsigned int refinement_l
       pcout<<"Used tolerance is: "<<tolerance<<endl;
       for (unsigned int i=0; i<cad_surfaces.size(); ++i)
         {
-          OpenCASCADE::NormalToMeshProjectionBoundary<2,3> normal_to_mesh_projector(cad_surfaces[i], tolerance);
-
-          normal_to_mesh_projectors.push_back(normal_to_mesh_projector);
+          normal_to_mesh_projectors.push_back(std::shared_ptr<OpenCASCADE::NormalToMeshProjectionBoundary<2,3> >
+                                    (new OpenCASCADE::NormalToMeshProjectionBoundary<2,3>(cad_surfaces[i], tolerance)));
```

Comment:
Thanks Alberto.
It does compile, but it doesn't work as it should. After all the computation is done, the program dies throwing an exception 

 --------------------------------------------------------
An error occurred in line <128> of file </home/amola/DEAL_II/dealii/source/base/subscriptor.cc> in function
    void dealii::Subscriptor::check_no_subscribers() const
The violated condition was: 
    counter == 0
The name and call sequence of the exception was:
    ExcInUse (counter, object_info->name(), infostring)
Additional Information: 
Object of class N6dealii11OpenCASCADE31ArclengthProjectionLineManifoldILi2ELi3EEE is still used by 1 other objects.

(Additional information: <none>)

So, it seems the shared pointer isn't helping here. I guess there's some other reason why the projector is still used when the triangulation dies, even if all the manifolds are indeed freed in the computational_domain class destructor